### PR TITLE
Python 3 compatiblity

### PR DESCRIPTION
**Pull request by marscher**

### CODE REVIEW

**Comment by marscher** [comment]
Coveralls is broken due to building via conda recipe.

**Comment by marscher** [comment]
Any volunteers to review this?

**Comment by franknoe** [comment]
I'm not an expert in Py3k. Maybe Moritz can review this?

**Comment by clonker** [comment]
I'll look through it tonight!

**Comment by marscher** [comment]
thank you!

File: pyemma/util/log.py [hunk 1]
```diff
@@ -28,10 +28,15 @@
...
 '''
+
+from __future__ import absolute_import
 __all__ = ['getLogger', 'enabled', 'CRITICAL', 'DEBUG', 'FATAL', 'INFO', 'NOTSET',
            'WARN', 'WARNING']
-
+from six import PY3
 import logging
+if PY3:
+    from imp import reload
```
Comment (clonker) [inline pyemma/util/log.py]: probably use

``` python
from importlib import reload
```

since imp.reload is deprecated

**Comment by clonker** [comment]
Not part of the changes in this PR but i stumbled upon it and its maybe still noteworthy and easily fixable: In msm/models/hmsm.py, lines 249, 265, 281, 296 and 311:

``` python
raise ValueError('observable vector a has size ' + len(a) + ' which is incompatible with both hidden (' +
                             self.nstates + ') and observed states (' + self.nstates_obs + ')')
```

.. this can never raise a meaningful error since len(a) is not a string type

File: pyemma/msm/estimators/lagged_model_validators.py [hunk 2]
```diff
@@ -1,10 +1,12 @@
+from __future__ import absolute_import
+from six.moves import range
 __author__ = 'noe'
 
 import numpy as np
 from pyemma._base.estimator import Estimator, estimate_param_scan, param_grid
 from pyemma._base.model import SampledModel
 #from pyemma._base.progress import ProgressReporter
-from pyemma._ext.six.moves import range
+from six.moves import range
```
Comment (clonker) [inline pyemma/msm/estimators/lagged_model_validators.py]: unnecessary double import of six.moves.range here

File: pyemma/msm/estimators/bayesian_msm.py [hunk 3]
```diff
@@ -5,7 +7,7 @@
 from pyemma.msm.models.msm_sampled import SampledMSM as _SampledMSM
 from pyemma.util.types import ensure_dtraj_list
 from pyemma._base.progress import ProgressReporter
-from pyemma._ext.six.moves import range
+from six.moves import range
```
Comment (clonker) [inline pyemma/msm/estimators/bayesian_msm.py]: unnecessary double import

File: pyemma/msm/estimators/bayesian_hmsm.py [hunk 4]
```diff
@@ -9,7 +11,7 @@
 from pyemma.msm.models.hmsm_sampled import SampledHMSM as _SampledHMSM
 from pyemma.util.units import TimeUnit
 from pyemma._base.progress import ProgressReporter
-from pyemma._ext.six.moves import range
+from six.moves import range
```
Comment (clonker) [inline pyemma/msm/estimators/bayesian_hmsm.py]: unnecessary double import

File: pyemma/datasets/api.py [hunk 5]
```diff
@@ -1,7 +1,8 @@
+from __future__ import (print_function, absolute_import)
```
Comment (clonker) [inline pyemma/datasets/api.py]: i don't think the print_function import is needed here

File: pyemma/coordinates/api.py [hunk 6]
```diff
@@ -59,8 +61,13 @@
 # types
 from mdtraj import Topology as _Topology, Trajectory as _Trajectory
 
+from six import string_types
+from six.moves import zip
+
 import numpy as _np
 import itertools as _itertools
+from six.moves import range
+from six.moves import zip
```
Comment (clonker) [inline pyemma/coordinates/api.py]: unnecessary double import of six.moves.zip

File: pyemma/coordinates/tests/__init__.py [hunk 7]
```diff
@@ -1,3 +1,4 @@
+from __future__ import absolute_import
```
Comment (clonker) [inline pyemma/coordinates/tests/__init__.py]: this should probably be below the copyright

File: pyemma/coordinates/data/writer.py [hunk 8]
```diff
@@ -71,7 +73,7 @@ def _reset(self, stride=1):
             pass
 
         try:
-            self._fh = open(self.filename, 'w')
+            self._fh = open(self.filename, 'wb')
```
Comment (clonker) [inline pyemma/coordinates/data/writer.py:76]: im not quite sure about this but isn't 'wb' binary mode and CSV is not binary..? I have found [here](http://stackoverflow.com/a/1020077/2871028) that "wb" is correct for python 2.x but "w" should be used for py3k when dealing with csv.

File: pyemma/_base/estimator.py [hunk 9]
```diff
@@ -1,3 +1,5 @@
+from __future__ import absolute_import
+from six.moves import range
```
Comment (clonker) [inline pyemma/_base/estimator.py:2]: unnecessary double import

**Pull request merged**